spec/rubykassa/xml_interface_spec.rb

```diff
@@ -31,4 +31,42 @@
   it "correctly transforms method name" do
     expect(subject.send(:transform_method_name, "some_method_name")).to eq("SomeMethodName")
   end
+
+  describe '#base_url' do
+    subject { described_class.new.base_url }
+    context 'sets to test URL when in test mode' do
+      before { Rubykassa.configure { |c| c.mode = :test } }
+      it { is_expected.to eq("http://test.robokassa.ru/Webservice/Service.asmx/") }
+    end
+
+    context 'uses production URL on production' do
+      before { Rubykassa.configure { |c| c.mode = :production } }
+      it { is_expected.to eq("https://merchant.roboxchange.com/WebService/Service.asmx/") }
+    end
+  end
+
+  describe '#op_state' do
+    subject { described_class.new.op_state(params) }
+
+    context 'depends on mode' do
+      let(:params) { { 'StateCode' => 5 } }
+
+      before do
+        allow(Net::HTTP).to receive(:get_response) do |params|
+          OpenStruct.new(code: '200', body: params.to_s)
+        end
+        allow(MultiXml).to receive(:parse) { |object| object }
+      end
+
+      context 'accepts additional params with test mode' do
+        before { Rubykassa.configure { |c| c.mode = :test } }
+        it { is_expected.to eq("http://test.robokassa.ru/Webservice/Service.asmx/OpState?MerchantLogin=your_login&InvoiceID=&Signature=7b7d63d7d72c05ee8470f467bda8adf1&StateCode=5") }
+      end
+
+      context 'ignores additional params with production mode' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.